modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPFailureTest.java

```diff
@@ -179,4 +199,248 @@ public void testPatchIdPNonExistentProperties() throws IOException {
+//        body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
+//                userDefinedAuthenticatorPayload.convertToJasonPayload());
+//        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+//        response.then()
+//                .log().ifValidationFails()
+//                .assertThat()
+//                .statusCode(FIELD_NOT_FOUND)
+//                .header(HttpHeaders.LOCATION, notNullValue());
+//
+//        String location = response.getHeader(HttpHeaders.LOCATION);
+//        assertNotNull(location);
+//        String customIdPId = location.substring(location.lastIndexOf("/") + 1);
+//        assertNotNull(customIdPId);
+//    }
+
+    // TODO: check 400
+//    @Test
+//    public void testAddIdPWithUserDefinedAuthenticatorWhenAuthenticatorAuthDetailsIsEmpty() throws IOException {
+//
+//        UserDefinedAuthenticatorPayload userDefinedAuthenticatorPayload =
+//                createUserDefinedAuthenticatorPayloadWithEmptyEndpointConfig("UmFuZG9tU3RyaW5nR2VuZXJhdG9y");
+//
+//        String body = idpCreatePayload.replace(FEDERATED_AUTHENTICATOR_ID_PLACEHOLDER,
+//                userDefinedAuthenticatorPayload.getAuthenticatorId());
+//        body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
+//                userDefinedAuthenticatorPayload.convertToJasonPayload());
+//        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+//        response.then()
+//                .log().ifValidationFails()
+//                .assertThat()
+//                .statusCode(HttpStatus.SC_CREATED)
+//                .header(HttpHeaders.LOCATION, notNullValue());
+//
+//        String location = response.getHeader(HttpHeaders.LOCATION);
+//        assertNotNull(location);
+//        String customIdPId = location.substring(location.lastIndexOf("/") + 1);
+//        assertNotNull(customIdPId);
+//    }
+
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticatorWhenAuthenticatorNameIsDuplicated() throws IOException {
+
+        userDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayload("Y3VzdG9tQXV0aGVudGljYXRvcjE=",
+                        "https://abc.com/authenticate",
+                        "testUser",
+                        "testPassword");
+        Response response = createUserDefAuthenticator("CustomAuthIDP1", userDefinedAuthenticatorPayload);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        customIdPId = location.substring(location.lastIndexOf("/") + 1);
+
+        // duplicate the authenticator creation
+        duplicatedUserDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayload("Y3VzdG9tQXV0aGVudGljYXRvcjE=",
```

Comment:
Lets make this string constants as we are using them mutiple places.

Reply:
This is updated with d6716dba397569d02ce08f1776ca188a75c02680